python/visualizer/ntupleviz.py

```diff
@@ -137,75 +149,124 @@ class TemplateDialog(FloatLayout):
+                channel_name = channel_name + '\nred=x green=y, blue=z'
+            elif data_type == 'point3':
+                visualiser = VisualiserPoint3(data_dict[data_type])
+                settings[data_type] = {'perspective': {},
+                                       'yaw': {},
+                                       'pitch': {}}
+                settings[data_type]['perspective'] = {'type': 'boolean',
+                                                      'default': True
+                                                      }
+                settings[data_type]['yaw'] = {'type': 'range',
+                                              'default': 0,
+                                              'min': -90,
+                                              'max': 90,
+                                              'step': 1
+                                              }
+                settings[data_type]['pitch'] = {'type': 'range',
+                                                'default': 0,
+                                                'min': -90,
+                                                'max': 90,
+                                                'step': 1
+                                                }
+            elif data_type == 'boundingBoxes':
+                visualiser = VisualiserBoundingBoxes(data_dict[data_type])
+                settings[data_type] = {'with_labels': {}}
+                settings[data_type]['with_labels'] = {'type': 'boolean',
+                                                      'default': True
+                                                      }
+            elif data_type == 'flowMap':
+                visualiser = VisualiserOpticFlow(data_dict[data_type])
+            elif data_type == 'mask':
+                visualiser = VisualiserMask(data_dict[data_type])
             else:
-                new_viewer = ViewerFrame()
-            visualiser = VisualiserFrame(data_dict)
-        elif data_type == 'pose6q':
-            new_viewer = ViewerPose6q()
-            visualiser = VisualiserPose6q(data_dict)
-            label = 'red=x green=y, blue=z ' + label
-        elif data_type == 'point3':
-            new_viewer = ViewerPoint3()
-            visualiser = VisualiserPoint3(data_dict)
-        new_viewer.label = label
-        new_viewer.dsm = visualiser
+                print("Warning! {} is not a recognized data type. Ignoring.".format(data_type))
+                continue
+            visualisers.append(visualiser)
+        new_viewer.title = channel_name
+        new_viewer.visualisers = visualisers
+        new_viewer.settings = settings
         self.add_widget(new_viewer)
 
         self.cols = int(np.ceil(np.sqrt(len(self.children))))
 
-    def add_viewer_for_each_channel_and_data_type(self, in_dict, label='', recursionDepth=0):
+    def add_viewer_for_each_channel_and_data_type(self, in_dict, seen_keys=None, recursionDepth=0):
```

Comment:
... or:

 def add_viewer_for_each_channel_and_data_type(self, in_dict, seen_keys=[], recursionDepth=0):

?

Reply:
doing as you suggest is a problem due to the mutable nature of lists, so you might incur in unwanted behaviours hard to debug